sklearnex/linear_model/incremental_linear.py

```diff
@@ -0,0 +1,392 @@
+            self.batch_size_ = 5 * n_features
+        else:
+            self.batch_size_ = self.batch_size
+
+        self.n_samples_seen_ = 0
+        if hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator._reset()
+
+        for batch in gen_batches(X.shape[0], self.batch_size_):
+            X_batch, y_batch = X[batch], y[batch]
+            self._onedal_partial_fit(X_batch, y_batch, queue=queue)
+
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        # finite check occurs on onedal side
+        self.n_features_in_ = X.shape[1]
+
+        if X.shape[0] == 1:
+            warnings.warn(
+                "Only one sample available. You may want to reshape your data array"
+            )
+
+        self._onedal_finalize_fit()
+
+        return self
+
+    def get_intercept_(self):
+        if hasattr(self, "_onedal_estimator"):
+            if self._need_to_finalize:
+                self._onedal_finalize_fit()
+
+            return self._onedal_estimator.intercept_
+        else:
+            raise AttributeError(
+                f"'{self.__class__.__name__}' object has no attribute 'intercept_'"
+            )
+
+    def set_intercept_(self, value):
+        self.__dict__["intercept_"] = value
+        if hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator.intercept_ = value
+            del self._onedal_estimator._onedal_model
+
+    def get_coef_(self):
+        if hasattr(self, "_onedal_estimator"):
+            if self._need_to_finalize:
+                self._onedal_finalize_fit()
+
+            return self._onedal_estimator.coef_
+        else:
+            raise AttributeError(
+                f"'{self.__class__.__name__}' object has no attribute 'coef_'"
+            )
+
+    def set_coef_(self, value):
+        self.__dict__["coef_"] = value
+        if hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator.coef_ = value
+            del self._onedal_estimator._onedal_model
+
+    coef_ = property(get_coef_, set_coef_)
+    intercept_ = property(get_intercept_, set_intercept_)
+
+    def partial_fit(self, X, y):
+        """
+        Incremental fit linear model with X and y. All of X and y is
+        processed as a single batch.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
+            Target values, where `n_samples` is the number of samples and
+            `n_targets` is the number of targets.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+
+        dispatch(
+            self,
+            "partial_fit",
+            {
+                "onedal": self.__class__._onedal_partial_fit,
+                "sklearn": None,
```

Comment:
What will happen in case if we are not supporting it via onedal branch?
It is undefined for me. This not the scope of current PR, but we should raise common error for such estimators, and some common test shoul check this error raised for such estimators.

Reply:
I guess it fails in dispatch in such cases. I will create task to handle it later.